### PR TITLE
Update rules_go to 0.16.0

### PR DESCRIPTION
**Pull request by katre**
Needed as part of bazelbuild/bazel#6283.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -147,9 +147,9 @@ groovy_repositories()
 # For our go_image test.
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "50207a04b74fe30218b06ac4467ea3862b9a3c99d3df7686be0eae02108ea06f",
+    sha256 = "ee5fe78fe417c685ecb77a0a725dc9f6040ae5beb44a0ba4ddb55453aad23a8a",
     strip_prefix = "rules_go-0.13.0",
```
Comment (xingao267) [inline WORKSPACE]: This should be updated as well.

Comment (katre) [reply WORKSPACE]: Thanks for catching that!

**Comment by xingao267** [comment]
Let's change the content in test_digest_output2 in tests/docker/BUILD to "sha256:8c70e530d7d9fd0018245579af25277d3ba0d47212b092225c391c41a82debef", this should be fixed properly in https://github.com/bazelbuild/rules_docker/pull/549

**Comment by katre** [comment]
Updated the SHA in the test.

**Pull request merged**